libraries/conf.rb

```diff
@@ -4,7 +4,35 @@
 # File Name:: conf.rb
 
 module CernerSplunk
+  # Utilities for working with splunk configuration files
   module Conf
+    def self.parse_string(conf)
+      return {} if conf.nil?
+      parse StringIO.open(conf, 'r:UTF-8')
+    end
+
+    def self.parse(io)
+      start = {}
+      hash = { 'default' => start }
+      current = start
+
+      io.each_line do |line|
+        case line
+        when /^\s*\[([^\]]+)\].*$/
+          start = {}
+          current = start
+          hash[Regexp.last_match[1].strip] = start
+        when /^\s*#.*$/
+          # ignore comment lines
+        when /^([^=]+)=(.*)$/
+          current[Regexp.last_match[1].strip] = Regexp.last_match[2].strip
```

Comment:
Maybe I'm splitting hairs but I'd prefer if these regexes where defined as a compiled constant and then used instead of relying on the state of `Regex.last_match`. 

Hopefully that makes sense.